lnwallet/witnessgen.go

```diff
@@ -0,0 +1,60 @@
+package lnwallet
+
+import (
+	"fmt"
+
+	"github.com/roasbeef/btcd/txscript"
+	"github.com/roasbeef/btcd/wire"
+)
+
+// WitnessType determines how an output's witness will be generated. The
+// default commitmentTimeLock type will generate a witness that will allow
+// spending of a time-locked transaction enforced by CheckSequenceVerify.
+type WitnessType uint16
+
+const (
+	// CommitmentTimeLock is a witness that allows us to spend the output of a
+	// commitment transaction after a relative lock-time lockout.
+	CommitmentTimeLock WitnessType = 0
```

Comment:
Ended up reverting this so that each witness type has a concrete value, this seemed to be the simplest approach to ensuring that ordering didn't corrupt any persistent state. Might try to fully decouple these in my next PR or in the process of packaging up various subsystems in main.